src/outputs/outputs.hpp

```diff
@@ -109,8 +109,10 @@ class OutputType {
   void CalculateCartesianVector(ParArrayND<Real> &src, ParArrayND<Real> &dst,
                                 Coordinates *pco);
   // following pure virtual function must be implemented in all derived classes
-  virtual void WriteOutputFile(Mesh *pm, ParameterInput *pin, bool flag) = 0;
-  virtual void WriteContainer(Mesh *pm, ParameterInput *pin, bool flag) { return; }
+  virtual void WriteOutputFile(Mesh *pm, ParameterInput *pin, SimTime *tm) = 0;
+  virtual void WriteContainer(SimTime &tm, Mesh *pm, ParameterInput *pin, bool flag) {
+    return;
+  }
```

Comment:
Since you are already changing this, it might make sense to make all the arguments const, I can't think of any reason why writing the contents to a file should change them.

Reply:
They shouldn't, but this isn't quite as simple as just adding the const qualifier to all the arguments.  Doing that makes the compiler barf because, for example, calling member functions in the ParameterInput object discards the const qualifier.  Let's put this off.